src/Ppu.re

```diff
@@ -0,0 +1,164 @@
+type registers = {
+  mutable control: int,
+  mutable mask: int,
+  mutable status: int,
+  mutable oam_address: int,
```

Comment:
Aside - I was thinking about adding something like 

```
type address = int
type byte = int
// others?
```

to `Types.re`. I'm not sure if there's a type-aliasing mechanism that would prevent you from inadvertently assigning one to the other, but I think it'd still be nice to be able to declare your intent a little more precisely.

Reply:
I definitely hear this. My frustration is that I want types that can constrict input to certain ranges and I feel the lack of such types is visible in all the places that we try to wrap around, i.e. `something land 0x1f`. We could create named types for our different address spaces like:
```
type PaletteIndex
type OAMIndex
type NametableIndex
type PRGIndex
type CHRIndex
etc
```
and then have them wrap during construction but that also feels maybe a bit heavyweight? I dunno. It does seem like something we'll have time to revisit.